sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,445 @@
+    private static final String CONTENT_TRANSFER_ENCODING = "Content-Transfer-Encoding: binary";
+    private static final String CONTENT_ID_TEMPLATE = "Content-ID: %d";
+    private static final String HTTP_VERSION = "HTTP/1.1";
+    private static final String OPERATION_TEMPLATE = "%s %s %s";
+    private static final String HEADER_TEMPLATE = "%s: %s";
+    private static final String NEWLINE = "\r\n";
+
+    private static final int[] EXPECTED_DELETE_STATUS_CODES = { 202 };
+    private static final int[] EXPECTED_SET_TIER_STATUS_CODES = { 200, 202 };
+
+    private final ClientLogger logger = new ClientLogger(BlobBatch.class);
+
+    private final BlobAsyncClient batchClient;
+
+    private final Deque<Mono<? extends Response<?>>> batchOperationQueue;
+    private final List<ByteBuffer> batchRequest;
+    private final Map<Integer, BlobBatchOperationResponse<?>> batchMapping;
+
+    private final AtomicInteger contentId;
+    private final String batchBoundary;
+
+    private BlobBatchType batchType;
+
+    /**
+     * Constructs a {@link BlobBatch} using the {@link BlobServiceClient#getHttpPipeline()
+     * BlobServiceClient's HttpPipeline} to build a modified {@link HttpPipeline} that is used to prepare the requests
+     * in the batch.
+     *
+     * @param client {@link BlobServiceClient} used to construct the batch.
+     */
+    public BlobBatch(BlobServiceClient client) {
+        this(client.getAccountUrl(), client.getHttpPipeline());
+    }
+
+    /**
+     * Constructs a {@link BlobBatch} using the {@link BlobServiceAsyncClient#getHttpPipeline()
+     * BlobServiceAsyncClient's HttpPipeline} to build a modified {@link HttpPipeline} that is used to prepare the
+     * requests in the batch.
+     *
+     * @param client {@link BlobServiceAsyncClient} used to construct the batch.
+     */
+    public BlobBatch(BlobServiceAsyncClient client) {
+        this(client.getAccountUrl(), client.getHttpPipeline());
+    }
+
+    BlobBatch(String accountUrl, HttpPipeline pipeline) {
+        this.contentId = new AtomicInteger(0);
+        this.batchBoundary = String.format(BATCH_BOUNDARY_TEMPLATE, UUID.randomUUID());
+
+        boolean batchHeadersPolicySet = false;
+        HttpPipelineBuilder batchPipelineBuilder = new HttpPipelineBuilder().httpClient(this::setupBatchOperation);
+        for (int i = 0; i < pipeline.getPolicyCount(); i++) {
+            HttpPipelinePolicy policy = pipeline.getPolicy(i);
+
+            if (policy instanceof SharedKeyCredentialPolicy) {
+                batchHeadersPolicySet = true;
+                // The batch policy needs to be added before the SharedKey policy to run preparation cleanup.
+                batchPipelineBuilder.policies(this::cleanseHeaders);
+            }
+
+            batchPipelineBuilder.policies(pipeline.getPolicy(i));
+        }
+
+        if (!batchHeadersPolicySet) {
+            batchPipelineBuilder.policies(this::cleanseHeaders);
```

Comment:
It doesn't look like you ever call build? Perhaps line 140 is supposed to pass batchPipelineBuilder.build() instead of pipeline?

Reply:
Fixed